include/cantera/thermo/PengRobinson.h

```diff
@@ -0,0 +1,313 @@
+
+    //! Update the a, b and alpha parameters
+    /*!
+     *  The a and the b parameters depend on the mole fraction and the
+     *  parameter alpha depends on the temperature. This function updates
+     *  the internal numbers based on the state of the object.
+     */
+    virtual void updateMixingExpressions();
+
+    //! Calculate the a, b and the alpha parameters given the temperature
+    /*!
+     * This function doesn't change the internal state of the object, so it is a
+     * const function.  It does use the stored mole fractions in the object.
+     *
+     * @param aCalc (output)  Returns the a value
+     * @param bCalc (output)  Returns the b value.
+     * @param aAlpha (output) Returns the (a*alpha) value.
+     */
+    void calculateAB(double& aCalc, double& bCalc, double& aAlpha) const;
+
+    void calcCriticalConditions(double& pc, double& tc, double& vc) const;
+
+    //! Prepare variables and call the function to solve the cubic equation of state
+    int solveCubic(double T, double pres, double a, double b, double aAlpha,
+                   double Vroot[3]) const;
+protected:
+    //! Value of b in the equation of state
+    /*!
+     *  m_b is a function of the mole fractions and species-specific b values.
+     */
+    double m_b;
+
+    //! Value of a and alpha in the equation of state
+    /*!
+     *  m_aAlpha_mix is a function of the temperature and the mole fractions. m_a depends only on the mole fractions.
+     */
+    double m_a;
+    double m_aAlpha_mix;
+
+    // Vectors required to store a_coeff, b_coeff, alpha, kappa and other values for every species. Length = m_kk
+    vector_fp m_b_coeffs;
+    vector_fp m_kappa;
+    mutable vector_fp m_dalphadT;
+    mutable vector_fp m_d2alphadT2;
+    vector_fp m_alpha;
+
+    //Matrices for Binary coefficients a_{i,j} and {a*alpha}_{i.j} are saved in an Array form. Length =  (m_kk, m_kk)
+    Array2D m_a_coeffs;
+    Array2D m_aAlpha_binary;
+
+    int m_NSolns;
+
+    double m_Vroot[3];
+
+    //! Temporary storage - length = m_kk.
+    mutable vector_fp m_pp;
+
+    // Partial molar volumes of the species
+    mutable vector_fp m_partialMolarVolumes;
+
+    //! The derivative of the pressure with respect to the volume
+    /*!
+     * Calculated at the current conditions. temperature and mole number kept
+     * constant
+     */
+    mutable double m_dpdV;
+
+    //! The derivative of the pressure with respect to the temperature
+    /*!
+     *  Calculated at the current conditions. Total volume and mole number kept
+     *  constant
+     */
+    mutable double m_dpdT;
+
+    //! Vector of derivatives of pressure with respect to mole number
+    /*!
+     *  Calculated at the current conditions. Total volume, temperature and
+     *  other mole number kept constant
+     */
+    mutable vector_fp m_dpdni;
+
+private:
+    //! Omega constants: a0 (= omega_a) and b0 (= omega_b) values used in Peng-Robinson equation of state
```

Comment:
This comment got associated with the `omega_a` documentation, but it doesn't seem to be related just to that one. Can you rephrase it?

Reply:
Corrected.